widlparser/constructs.py

```diff
@@ -468,6 +469,7 @@ def __init__(self, tokens, parent = None, parser = None):
         Construct.__init__(self, tokens, parent, (not parent), parser = parser)
         self.partial = Symbol(tokens, 'partial') if (Symbol.peek(tokens, 'partial')) else None
         self._interface = Symbol(tokens, 'interface')
+        self.mixin = Symbol(tokens, 'mixin') if (Symbol.peek(tokens, 'mixin')) else None

```

Comment:
There is a difference in the set of children allowed [in an interface](https://heycam.github.io/webidl/#prod-InterfaceMember) versus [in an interface mixin](https://heycam.github.io/webidl/#prod-MixinMember). Specifically, the following are not allowed in interface mixins:

- [Special operations](https://heycam.github.io/webidl/#prod-SpecialOperation) (stringifiers declared on either an operation or an attribute are fine though)
- [Static members](https://heycam.github.io/webidl/#prod-StaticMember)
- [Maplike declarations](https://heycam.github.io/webidl/#prod-ReadWriteMaplike) (readonly or not)
- [Setlike declarations](https://heycam.github.io/webidl/#prod-ReadWriteSetlike) (readonly or not)
- [Inherited attributes](https://heycam.github.io/webidl/#prod-ReadWriteAttribute)

Currently this is not enforced.